src/api/spec/helpers/webui/webui_helper_spec.rb

```diff
@@ -319,7 +319,69 @@
   end
 
   describe '#user_and_role' do
-    skip('Please add some tests')
+    let(:user) { create(:user) }
+    let(:nobody) { create(:user_nobody) }
+
+    before do
+      User.current = user
+    end
+
+    context 'for users' do
+      it 'display the users realname with its login' do
+        expect(user_and_role(user.login)).to eq(
+          "<img width=\"20\" height=\"20\" alt=\"#{user.realname}\" src=\"/user/icon/#{user.login}?size=20\" " + \
+          "/><a href=\"/user/show/#{user.login}\">#{user.realname} (#{user.login})</a>"
+        )
+      end
+
+      it 'falls back to users login if realname is empty' do
+        user.realname = ''
+        user.save!
+
+        expect(user_and_role(user.login)).to eq(
+          "<img width=\"20\" height=\"20\" alt=\"#{user}\" src=\"/user/icon/#{user.login}?size=20\" " + \
+          "/><a href=\"/user/show/#{user.login}\">#{user.login}</a>"
+        )
+      end
+    end
+
+    context 'with options' do
+      it 'does not show an icon if option disables it' do
+        expect(user_and_role(user.login, nil, no_icon: true)).to eq(
+          "<a href=\"/user/show/#{user.login}\">#{user.realname} (#{user.login})</a>"
+        )
+      end
+
+      it 'only shows user login if short option is set' do
+        user.realname = ''
+        user.save!
```

Comment:
You would get
```
       expected: "<img width=\"20\" height=\"20\" alt=\"user_1\" src=\"/user/icon/user_1?size=20\" />user_1"
            got: "<img width=\"20\" height=\"20\" alt=\"Seamus Baumbach Sr.\" src=\"/user/icon/user_1?size=20\" />user_1"
```
otherwise.